### PR TITLE
S3 Job Queue

### PR DESCRIPTION
**Pull request by mariusandra**
## Changes

- Still WIP
- Implements a simple redlocked job queue based on S3. 
- Needs tests
- Closes https://github.com/PostHog/plugin-server/issues/364

## Checklist

-   [ ] Updated Settings section in README.md, if settings are affected
-   [ ] Jest tests

### CODE REVIEW

**Comment by mariusandra** [comment]
Here's something that could use a look. :)

I grouped the different job queues into folders based on functionality (local, concurrent, redlocked), but didn't yet implement different functionalities depending on the type. This means currently all job queues are still redlocked. I'd suggest tackling this in another PR.

**Review by Twixes — APPROVED** [review]
Looks reasonable to me

File: src/main/job-queues/redlocked/s3-queue.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        await this.connectS3()
+    }
+
+    async enqueue(retry: EnqueuedJob): Promise<void> {
+        if (!this.s3Wrapper) {
+            throw new Error('S3 object not initialized')
+        }
+        const date = new Date(retry.timestamp).toISOString()
+        const [day, time] = date.split('T')
+        const dayTime = `${day.split('-').join('')}-${time.split(':').join('')}`
```
Comment (neilkakkar) [inline src/main/job-queues/redlocked/s3-queue.ts:37]: was there a specific reason for formatting things like this?

Comment (mariusandra) [reply src/main/job-queues/redlocked/s3-queue.ts:37]: Basically just keeping it sortable to the second while removing special symbols that might interfere with it being a filename.

**Pull request merged**